controllers/keystoneapi_controller.go

```diff
@@ -229,6 +244,11 @@ func (r *KeystoneAPIReconciler) reconcileInit(
 		ctx,
 		helper,
 	)
+	// Register the finalizer immediately to avoid orphaning resources on delete
+	if err := r.Update(ctx, instance); err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
we do not need this. you create/update the db object in db.CreateOrPatchDB and you set the finalizer there

Reply:
ok done